### PR TITLE
sticking WebService in context [WIP]

### PR DESCRIPTION
**Pull request by inkblotty**
- moving WebService into new React 16 context
- helper methods for wrapping in context and retrieving from context
** Thoughts, comments, concerns appreciated **

### CODE REVIEW

File: src/js/components/app/app-container.jsx [hunk 1]
```diff
@@ -79,29 +85,35 @@ class AppContainer extends React.Component {
...
-        <MainRouter />
-        <Footer />
-        <Alert currentAlert={alerts[0]} hidden={!alerts[0]} />
-        <Modal
-          modalContents={modal && modal.modalContents}
-          modalTriggerId={modal && modal.modalTriggerId}
-          closeModal={closeModal}
-        />
-      </div>
+      <AppContext.Provider value={{ WebService: AppWebService }}>
```
Comment (inkblotty) [inline src/js/components/app/app-container.jsx:93]: Context has to be shoved into a Provider, similarly to Redux.

File: src/js/components/home/home-container.jsx [hunk 2]
```diff
@@ -65,4 +67,9 @@ const mapStateToProps = state => {
   }
 }
 
-export default connect(mapStateToProps, mapDispatchToProps)(HomeContainer)
+// example of how to connect a stateful component to context, with or without Redux
+
```
Comment (inkblotty) [inline src/js/components/home/home-container.jsx]: Example 1

File: src/js/components/navigation/header/header.jsx [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
       <Link to='/'>
-        <Logo full={true} />
+        <Logo full />
       </Link>
       <SubHeader openModal={props.openModal} user={props.user} />
     </header>
   )
 }
 
-export default withRouter(Header)
```
Comment (inkblotty) [inline src/js/components/navigation/header/header.jsx:21]: Example 2

File: src/js/redux-store/actions/userActions.js [hunk 4]
```diff
@@ -2,10 +2,12 @@ import actionTypes from './types'
 
 /* user action creators */
 
+import { getCurrentContext } from 'components/app/app-context'
```
Comment (inkblotty) [inline src/js/redux-store/actions/userActions.js:5]: Example: Getting context outside of React

**Review by deadpointdigital — DISMISSED** [review]

**Review by silentorb — DISMISSED** [review]

**Review by deadpointdigital — APPROVED** [review]

**Pull request merged**